app/models/miq_worker.rb

```diff
@@ -455,6 +455,18 @@ def clean_active_messages
     end
   end
 
+  def error_out_tasks_with_active_queue_message
+    message = "Task Handler: [#{friendly_name}] ID [#{id}] has been deleted!"
+    processed_messages.includes(:miq_task).each do |m|
+      task = m.miq_task
+      next unless task
+
+      # Note, this is done synchronously from destroy because workers have 1 message they're currently "handling"
+      # and each message can only have 1 task, so this should be very fast even for many workers.
+      task.update_status(MiqTask::STATE_FINISHED, MiqTask::STATUS_ERROR, message)
```

Comment:
```suggestion
      # Note, this is done synchronously from destroy because workers have 1 message they're currently "handling"
      # and each message can only have 1 task, so this should be very fast even for many workers.
      m.miq_task&.update_status(MiqTask::STATE_FINISHED, MiqTask::STATUS_ERROR, message)
```

Reply:
I considered something like this but kept this split out because most/many queue messages will not have a task associated with them.  With the guard clause, it seems expected, whereas `&.` it seems an exception.  Maybe that's just me?  

I guess I could also do `where.not(:miq_task_id => nil)` and not check the `task`

Reply:
I like the exclusion of the nil objects entirely, but that might incur an extra unneeded query - not sure.

Reply:
Yea, I like that exclude `where` clause.
I remember having a map method in preloader but just looked at it and gasped. Will try and put that tangent out of my mind.